astropy/io/ascii/daophot.py

```diff
@@ -165,15 +195,105 @@ class DaophotData(core.BaseData):
     start_line = 0
     comment = r'\s*#'
 
+    def __init__(self):
+        core.BaseData.__init__(self)
+        self.is_multiline = False
+
+    def get_data_lines(self, lines):
+
+        # Special case for multiline daophot databases. Extract the aperture values from the first multiline data block
+        if self.is_multiline:
+            #grab the first column of the special block (aperture values) and recreate the aperture description string
+            aplist = next( zip(*map(str.split, self.first_block)) )
+            self.header.aperture_values = tuple(map( float, aplist ))
+
+        # Set self.data.data_lines to a slice of lines contain the data rows
+        core.BaseData.get_data_lines(self, lines)
+
 
 class DaophotInputter(core.ContinuationLinesInputter):
-    no_continue = r'\s*#'
+
+    continuation_char = '\\'
+    multiline_char = '*'
+    replace_char = ' '
+    re_multiline     = re.compile( r'(#?)[^\\*#]*(\*?)(\\*) ?$' )
+
+
+    def search_multiline(self, lines, depth=150):
+        '''search lines for special continuation character to determine number of continued rows in a
+        datablock.  For efficiency, depth gives the upper limit of lines to search.'''
```

Comment:
Instinctively, I feel like it shouldn't have too great an effect on performance to increase this value of the depth parameter.  I can do a few tests to confirm this.  The databases I dealt with are typically a few hundred thousand lines, which means the majority of the time is spent parsing the main body of the database, not the header!